torch/csrc/distributed/c10d/nvshmem_extension.cu

```diff
@@ -418,4 +418,38 @@ at::Tensor nvshmem_reduce_scatter_out(
   return out;
 }
 
+#define THREADS_PER_BLOCK 512
+
+__global__ void sendrecv(float *send_data, float *recv_data, int num_elems, int mype,
+                                     int npes) {
+    int thread_idx = blockIdx.x * blockDim.x + threadIdx.x;
+    int peer = (mype + 1) % npes;
+    int block_offset = blockIdx.x * blockDim.x;
+    // All threads in a block call the API with the same arguments
+    nvshmemx_float_put_block(recv_data + block_offset, send_data + block_offset,
```

Comment:
I think it would be less confusing to use nvshmemx_putSIZE_block https://docs.nvidia.com/nvshmem/api/gen/api/rma.html#c.nvshmemx_putSIZE_block instead of using fake float * pointers
Also, I'm curious whether we need *block functions for cross-node transfers, where one thread should be enough?